src/samples/authentication/user_builder.ts

```diff
@@ -0,0 +1,37 @@
+import { Request } from 'express';
+import { UnauthenticatedUser, User } from '../../server/index.js';
+import { UserBuilder } from '../../server/express/common.js';
+
+export class CustomUser implements User {
+  constructor(
+    private _userName: string,
+    private _email: string,
+    private _role: string
+  ) {}
+
+  public isAuthenticated(): boolean {
+    return true;
+  }
+
+  public userName(): string {
+    return this._userName;
+  }
+
+  public email(): string {
+    return this._email;
+  }
+
+  public role(): string {
+    return this._role;
+  }
+}
```

Comment:
I'd simplify the declaration and put the content of this file together with authentication middleware to make it clear how things work together and minimize fragmentation.

```suggestion
export class CustomUser implements User {
  constructor(
    public userName: string,
    public email: string,
    public role: string
  ) {}

  public isAuthenticated(): boolean {
    return true;
  }
}
```